components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +180,51 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// findDistance calculates the distance between two points on Earth.
+// lat1, lon1: Latitude and Longitude of point 1 (in decimal degrees)
+// lat2, lon2: Latitude and Longitude of point 2 (in decimal degrees)
+// returns the distance in meters.
+func findDistance(lat1, lon1, lat2, lon2 float64) float64 {
```

Comment:
so... the `geo.Point` library we import has this helper function... please use that.

Example in dual GPS somewhere. Always check the library functions if you're importing a library for a specific type.